byterun/str.c

```diff
@@ -191,7 +206,16 @@ CAMLprim value caml_string_set16(value str, value index, value newval)
   return Val_unit;
 }
 
-CAMLprim value caml_string_set32(value str, value index, value newval)
+/**
+ * [caml_string_set16] is deprecated,
+ * use [caml_bytes_set16] instead
+ */
```

Comment:
What is the reason for keeping these `caml_string_set*` C primitives? AFAICS they are not exposed to user code and not used by the bytecode interpreter.

Reply:
I guess it is related to your previous comment, but someone might
reference them through `external`, *e.g.* [ocp-endian](https://github.com/OCamlPro/ocplib-endian/blob/master/src/endianString.cppo.ml#L94).

Reply:
But they would reference the `%caml_string_set*` name, not the C primitive, right?

Reply:
Sorry, you meant "make the old OCaml primitive name point to a new C symbol"?

I think there could still be a problem, with an `.o` file generated by a previous version
of the compiler and then linked with the new runtime; however, if magic numbers are
bumped it should not be a problem (cannot remember whether magic numbers are
automatically bumped nowadays).

Reply:
I've made the change.